frontend/public/components/instantiate-template.tsx

```diff
@@ -156,7 +156,7 @@ class TemplateForm_ extends React.Component<
   getParameterValues = (props: TemplateFormProps & WithTranslation) => {
```

Comment:
Looks like `componentDidUpdate` is resetting state, you might want to check why this function is being run there and avoid calling it if not needed.

Reply:
This function is run to extract the name and value of the parameters that are hardcoded for a particular Template Instance and put it inside the current state object. This function has been called twice in the file at different places.

I was able to resolve @invincibleJai 's comment by altering the sequence of the conditional or statement. This solved the problem.

Reply:
The fix works, but ideally we shouldn't be calling this function if there is no props update. We won't need to update the function's behavior if we can avoid calling it.

I see that this component is connected to firehose, so the first few renders will not have obj loaded. `componentDidUpdate` is used to set params when the resource is loaded. We need to avoid calling `getParameterValues` if the object has not been updated.

Reply:
Yes. Have fixed the code. Now the `getParameterValues` function is getting called only once.